token-swap/program/src/instruction.rs

```diff
@@ -29,13 +82,7 @@ pub enum SwapInstruction {
     ///   6. `[writable]` Pool Token Account to deposit the initial pool token
     ///   supply.  Must be empty, not owned by $authority.
     ///   7. '[]` Token program id
-    Initialize {
-        /// nonce used to create valid program address
-        nonce: u8,
-        /// swap curve info for pool, including CurveType, fees, and anything
-        /// else that may be required
-        swap_curve: SwapCurve,
-    },
+    Initialize(Initialize),
```

Comment:
Kinda sad to see this.  Arbitrary doesn't work over enums?

Reply:
I totally agree.  This is actually long-standing rust issue that enum variants don't count as types.  I didn't follow the whole discussion over here, but I know there were efforts to make that happen: https://github.com/rust-lang/rust/issues/40666

Reply:
the other option is to have the fuzz instructions live separately, but I preferred having them re-use the existing instructions.  I can be overruled on that though

Reply:
Separately sounds a lot worse.   Maybe there'll be other options when using Borsh encoding since there we have the ability to programmatically figure the types, even from an enum, and could use that information to construct the fuzzer inputs

Reply:
Oh yeah that's a great point.  I'll look into possibilities for how to integrate borsh with the libfuzzer inputs and ditch Arbitrary

Reply:
sweet, `feature-proposal` is already borshed up and its instruction interface is really small.  It'd be a good guinea pig for Borsh-based fuzzing.  (cc:  https://github.com/solana-labs/solana-program-library/blob/master/feature-proposal/program/src/borsh_utils.rs).   Might be slightly quicker to iterate on than token-swap for experiments.

Reply:
good point, I'll take a look at it next then :-)